hw/mcu/dialog/da1469x/src/da1469x_trimv.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <assert.h>
+#include <string.h>
+#include "mcu/da1469x_otp.h"
+#include "mcu/da1469x_trimv.h"
+
+/*
+ * XXX
+ * CS in OTP has 1024 bytes which is 256 words so in theory it could store up to
+ * 254 trim values (2 words are reserved for start and trim commands). Not sure
+ * if we really need to reserve that much space, perhaps this should be less by
+ * default or configurable. Largest group id found in OTP was 18 so far.
+ */
+#define TRIMV_COUNT_MAX         254
+#define TRIMV_GROUP_ID_MAX      18
+
+#define CS_WORD_START           0xa5a5a5a5
+#define CS_WORD_END             0x00000000
+#define CS_WORD_INVALID         0xffffffff
+#define CS_WORD_TYPE_MASK       0xf0000000
+#define CS_WORD_TYPE_BOOTER     0x60000000
+#define CS_WORD_TYPE_TRIM       0x90000000
+
+struct da1469x_trimv_group {
+    uint8_t start;
+    uint8_t length;
+};
+
+static uint32_t g_mcu_trimv[TRIMV_COUNT_MAX];
```

Comment:
I don't know how often trim values are needed, but this seems like a lot of RAM to use for duplicating data which is readily readable from OTP.
I would rather just store pointers to OTP in g_mcu_trimv_groups, and call otp_read() when da1469x_trimv_get_group() is getting called.

Reply:
that's a good idea, will fix. if code needs to add some trim values frequently, it can load these values to RAM - no need to load all values which we may not need at all.